Installation issue: ecflow@5.11.4 with boost@1.85.0

### Steps to reproduce the issue

```console
$ spack spec -I ecflow@5.11.4
Concretized
--------------------------------
 -   pmiihk7  ecflow@5.11.4%apple-clang@13.1.6~ipo~pic+ssl~static_boost+ui build_system=cmake build_type=Release generator=make patches=2e6d6d0 arch=darwin-monterey-m1
[+]  xputgpa      ^boost@1.85.0%apple-clang@13.1.6~atomic+chrono~clanglibcpp~container~context~contract~coroutine+date_time~debug+exception~fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale~log~math~mpi+multithreaded~nowide~numpy+pic+program_options+python~random+regex+serialization+shared~signals~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~versionedlayout~wave build_system=generic cxxstd=17 patches=a440f96,b8569d7 visibility=hidden arch=darwin-monterey-m1
[e]  woh4c35      ^cmake@3.27.0%apple-clang@13.1.6~doc+ncurses+ownlibs build_system=generic build_type=Release arch=darwin-monterey-m1
[e]  t7pdriq      ^gmake@3.81%apple-clang@13.1.6~guile build_system=generic patches=ca60bd9 arch=darwin-monterey-m1
[+]  56mxtnq      ^openssl@3.3.0%apple-clang@13.1.6~docs+shared build_system=generic certs=mozilla arch=darwin-monterey-m1
[+]  4fez47j          ^ca-certificates-mozilla@2023-05-30%apple-clang@13.1.6 build_system=generic arch=darwin-monterey-m1
[e]  cxwfivy          ^perl@5.36.1%apple-clang@13.1.6~cpanm+opcode+open+shared+threads build_system=generic arch=darwin-monterey-m1
[+]  pklv2l6          ^zlib-ng@2.1.6%apple-clang@13.1.6+compat+new_strategies+opt+pic+shared build_system=autotools arch=darwin-monterey-m1
[+]  qzxsoh5              ^gnuconfig@2022-09-17%apple-clang@13.1.6 build_system=generic arch=darwin-monterey-m1
[+]  p75vpq5      ^py-numpy@1.22.3%apple-clang@13.1.6 build_system=python_pip patches=873745d,e9508c3 arch=darwin-monterey-m1
[+]  tm26m7g          ^openblas@0.3.24%apple-clang@13.1.6~bignuma~consistent_fpcsr+dynamic_dispatch+fortran~ilp64+locking+noavx512+pic+shared build_system=makefile symbol_suffix=none threads=none arch=darwin-monterey-m1
[+]  eu7ymgb          ^py-cython@0.29.36%apple-clang@13.1.6 build_system=python_pip patches=c4369ad arch=darwin-monterey-m1
[+]  ajxr6wz          ^py-wheel@0.41.2%apple-clang@13.1.6 build_system=generic arch=darwin-monterey-m1
[+]  g2ur6w5      ^py-pip@23.1.2%apple-clang@13.1.6 build_system=generic arch=darwin-monterey-m1
[+]  ol2atfd      ^py-setuptools@63.4.3%apple-clang@13.1.6 build_system=generic arch=darwin-monterey-m1
[+]  qiflf6a      ^python@3.10.13%apple-clang@13.1.6+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,ebdca64,f2fd060 arch=darwin-monterey-m1
[e]  tt5g7ts          ^apple-libuuid@1353.100.2%apple-clang@13.1.6 build_system=bundle arch=darwin-monterey-m1
[+]  uzry4zw          ^bzip2@1.0.8%apple-clang@13.1.6~debug~pic+shared build_system=generic arch=darwin-monterey-m1
[e]  gij2sdt              ^diffutils@2.8.1%apple-clang@13.1.6 build_system=autotools arch=darwin-monterey-m1
[+]  esdvqma          ^expat@2.6.2%apple-clang@13.1.6~libbsd build_system=autotools arch=darwin-monterey-m1
[+]  akppsb2          ^gdbm@1.23%apple-clang@13.1.6 build_system=autotools arch=darwin-monterey-m1
[e]  py5wmv7          ^gettext@0.21.1%apple-clang@13.1.6+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=darwin-monterey-m1
[+]  lxvlye5          ^libffi@3.4.6%apple-clang@13.1.6 build_system=autotools arch=darwin-monterey-m1
[+]  ubja6wg          ^libxcrypt@4.4.35%apple-clang@13.1.6~obsolete_api build_system=autotools patches=4885da3 arch=darwin-monterey-m1
[+]  7mugmh2          ^ncurses@6.5%apple-clang@13.1.6~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=darwin-monterey-m1
[+]  sxnd7px          ^pkg-config@0.29.2%apple-clang@13.1.6+internal_glib build_system=autotools arch=darwin-monterey-m1
[+]  6b34edq          ^readline@8.2%apple-clang@13.1.6 build_system=autotools patches=bbf97f1 arch=darwin-monterey-m1
[+]  gtxwbog          ^sqlite@3.43.2%apple-clang@13.1.6+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=darwin-monterey-m1
[+]  sxdmjwc          ^xz@5.4.6%apple-clang@13.1.6~pic build_system=autotools libs=shared,static arch=darwin-monterey-m1
[+]  a7h6s6o      ^python-venv@1.0%apple-clang@13.1.6 build_system=generic arch=darwin-monterey-m1
[e]  lseduc7      ^qt@5.15.10%apple-clang@13.1.6~dbus~debug~doc~examples+framework~gtk+gui~opengl~phonon+shared+sql+ssl+tools~webkit build_system=generic patches=51aeba5,7f34d48,c2d75d9 arch=darwin-monterey-m1
```


### Error message

```
[  5%] Linking CXX static library libcore.a
cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /opt/homebrew/Cellar/cmake/3.27.0/bin/cmake -P CMakeFiles/core.dir/cmake_clean_target.cmake
cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /opt/homebrew/Cellar/cmake/3.27.0/bin/cmake -E remove /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore/libcore.a
cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /opt/homebrew/Cellar/cmake/3.27.0/bin/cmake -E cmake_link_script CMakeFiles/core.dir/link.txt --verbose=1
/usr/bin/ar qc libcore.a CMakeFiles/core.dir/src/AssertTimer.cpp.o CMakeFiles/core.dir/src/Cal.cpp.o CMakeFiles/core.dir/src/Calendar.cpp.o CMakeFiles/core.dir/src/Child.cpp.o CMakeFiles/core.dir/src/CommandLine.cpp.o CMakeFiles/core.dir/src/DState.cpp.o CMakeFiles/core.dir/src/DurationTimer.cpp.o CMakeFiles/core.dir/src/Ecf.cpp.o CMakeFiles/core.dir/src/Extract.cpp.o CMakeFiles/core.dir/src/File.cpp.o CMakeFiles/core.dir/src/File_r.cpp.o CMakeFiles/core.dir/src/Host.cpp.o CMakeFiles/core.dir/src/Indentor.cpp.o CMakeFiles/core.dir/src/Log.cpp.o CMakeFiles/core.dir/src/LogVerification.cpp.o CMakeFiles/core.dir/src/NOrder.cpp.o CMakeFiles/core.dir/src/NState.cpp.o CMakeFiles/core.dir/src/NodePath.cpp.o CMakeFiles/core.dir/src/Passwd.cpp.o CMakeFiles/core.dir/src/PasswdFile.cpp.o CMakeFiles/core.dir/src/Pid.cpp.o CMakeFiles/core.dir/src/PrintStyle.cpp.o CMakeFiles/core.dir/src/SState.cpp.o CMakeFiles/core.dir/src/Str.cpp.o CMakeFiles/core.dir/src/StringSplitter.cpp.o CMakeFiles/core.dir/src/TimeSeries.cpp.o CMakeFiles/core.dir/src/TimeSlot.cpp.o CMakeFiles/core.dir/src/TimeStamp.cpp.o CMakeFiles/core.dir/src/User.cpp.o CMakeFiles/core.dir/src/Version.cpp.o CMakeFiles/core.dir/src/WhiteListFile.cpp.o
/usr/bin/ranlib libcore.a
[  5%] Built target core
make: *** [all] Error 2
==> Error: ProcessError: Command exited with status 2:
    'make' '-j10'

24 errors found in build log:
     240
     241    Call Stack (most recent call first):
     242      Pyext/CMakeLists.txt:168 (ecbuild_find_python)
     243    This warning is for project developers.  Use -Wno-dev to suppress it.
     244
     245    -- Found PythonInterp: /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/envs/unified-env/install/apple-clang/13.1.6/python-venv-1.0-a7h6s6o/bin/python3 (found version "
            3.10.13")
  >> 246    <string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
  >> 247    <string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
     248    CMake Warning (dev) at cmake/ecbuild_find_python.cmake:203 (find_package):
     249      Policy CMP0148 is not set: The FindPythonInterp and FindPythonLibs modules
     250      are removed.  Run "cmake --help-policy CMP0148" for policy details.  Use
     251      the cmake_policy command to set the policy and suppress this warning.
     252
     253    Call Stack (most recent call first):

     ...

     480    [  4%] Building CXX object ACore/CMakeFiles/core.dir/src/NOrder.cpp.o
     481    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /Users/hei
            nzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/spack/lib/spack/env/clang/clang++ -DCMAKE -DECFLOW_LOGVIEW -DECFLOW_QT5 -DECF_OPENSSL -DQT_NO_KEYWORDS -I/Users/heinzell/work/spa
            ck-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/cereal/include -I/Users/heinzell/work/spack-stack/update
            _spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src -I/Users/heinzell/work/spack-stack/update_spack_stack_from_sp
            ack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7 -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spa
            ck-stack-merge-in-spack-develop/envs/unified-env/install/apple-clang/13.1.6/openssl-3.3.0-56mxtnq/include -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/envs/unifie
            d-env/install/apple-clang/13.1.6/boost-1.85.0-xputgpa/include -DBOOST_NO_CXX98_FUNCTION_BASE -pipe -O3 -DNDEBUG -std=gnu++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX.sdk -mmacosx-version-min=12.0 -fPIC -MD -MT ACore/CMakeFiles/core.dir/src/NOrder.cpp.o -MF CMakeFiles/core.dir/src/NOrder.cpp.o.d -o CMakeFiles/core.dir/src/NOrder.cpp.o -c /Users/heinzell/work/spack-stack/update_spack_sta
            ck_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src/NOrder.cpp
     482    [  4%] Building CXX object ACore/CMakeFiles/core.dir/src/NState.cpp.o
     483    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /Users/hei
            nzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/spack/lib/spack/env/clang/clang++ -DCMAKE -DECFLOW_LOGVIEW -DECFLOW_QT5 -DECF_OPENSSL -DQT_NO_KEYWORDS -I/Users/heinzell/work/spa
            ck-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/cereal/include -I/Users/heinzell/work/spack-stack/update
            _spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src -I/Users/heinzell/work/spack-stack/update_spack_stack_from_sp
            ack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7 -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spa
            ck-stack-merge-in-spack-develop/envs/unified-env/install/apple-clang/13.1.6/openssl-3.3.0-56mxtnq/include -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/envs/unifie
            d-env/install/apple-clang/13.1.6/boost-1.85.0-xputgpa/include -DBOOST_NO_CXX98_FUNCTION_BASE -pipe -O3 -DNDEBUG -std=gnu++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX.sdk -mmacosx-version-min=12.0 -fPIC -MD -MT ACore/CMakeFiles/core.dir/src/NState.cpp.o -MF CMakeFiles/core.dir/src/NState.cpp.o.d -o CMakeFiles/core.dir/src/NState.cpp.o -c /Users/heinzell/work/spack-stack/update_spack_sta
            ck_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src/NState.cpp
     484    [  4%] Building CXX object ACore/CMakeFiles/core.dir/src/NodePath.cpp.o
     485    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /Users/hei
            nzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/spack/lib/spack/env/clang/clang++ -DCMAKE -DECFLOW_LOGVIEW -DECFLOW_QT5 -DECF_OPENSSL -DQT_NO_KEYWORDS -I/Users/heinzell/work/spa
            ck-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/cereal/include -I/Users/heinzell/work/spack-stack/update
            _spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src -I/Users/heinzell/work/spack-stack/update_spack_stack_from_sp
            ack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7 -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spa
            ck-stack-merge-in-spack-develop/envs/unified-env/install/apple-clang/13.1.6/openssl-3.3.0-56mxtnq/include -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/envs/unifie
            d-env/install/apple-clang/13.1.6/boost-1.85.0-xputgpa/include -DBOOST_NO_CXX98_FUNCTION_BASE -pipe -O3 -DNDEBUG -std=gnu++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX.sdk -mmacosx-version-min=12.0 -fPIC -MD -MT ACore/CMakeFiles/core.dir/src/NodePath.cpp.o -MF CMakeFiles/core.dir/src/NodePath.cpp.o.d -o CMakeFiles/core.dir/src/NodePath.cpp.o -c /Users/heinzell/work/spack-stack/update_spa
            ck_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src/NodePath.cpp
  >> 486    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:80:45: error: no type named 'filesystem_error' in namespace 'boost::filesystem'
     487                catch (const boost::filesystem::filesystem_error& err) {
     488                             ~~~~~~~~~~~~~~~~~~~^
  >> 489    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:107:41: error: no type named 'filesystem_error' in namespace 'boost::filesystem'
     490            catch (const boost::filesystem::filesystem_error& err) {
     491                         ~~~~~~~~~~~~~~~~~~~^
  >> 492    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:180:45: error: no type named 'filesystem_error' in namespace 'boost::filesystem'
     493                catch (const boost::filesystem::filesystem_error& e) {
     494                             ~~~~~~~~~~~~~~~~~~~^
  >> 495    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:232:42: error: expected ';' after expression
     496        boost::filesystem::directory_iterator it(path), eod;
     497                                             ^
     498                                             ;
  >> 499    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:232:24: error: no member named 'directory_iterator' in namespace 'boost::filesystem'
     500        boost::filesystem::directory_iterator it(path), eod;
     501        ~~~~~~~~~~~~~~~~~~~^
  >> 502    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:232:43: error: use of undeclared identifier 'it'
     503        boost::filesystem::directory_iterator it(path), eod;
     504                                              ^
  >> 505    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:232:53: error: use of undeclared identifier 'eod'
     506        boost::filesystem::directory_iterator it(path), eod;
     507                                                        ^
  >> 508    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:69: error: use of undeclared identifier 'it'
     509        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     510                                                                        ^
  >> 511    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:73: error: use of undeclared identifier 'eod'
     512        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     513                                                                            ^
  >> 514    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:69: error: use of undeclared identifier 'it'
     515        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     516                                                                        ^
  >> 517    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:73: error: use of undeclared identifier 'eod'
     518        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     519                                                                            ^
  >> 520    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:69: error: use of undeclared identifier 'it'
     521        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     522                                                                        ^
  >> 523    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:73: error: use of undeclared identifier 'eod'
     524        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     525                                                                            ^
  >> 526    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:69: error: use of undeclared identifier 'it'
     527        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     528                                                                        ^
  >> 529    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:73: error: use of undeclared identifier 'eod'
     530        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     531                                                                            ^
  >> 532    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:69: error: use of undeclared identifier 'it'
     533        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     534                                                                        ^
  >> 535    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:73: error: use of undeclared identifier 'eod'
     536        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     537                                                                            ^
  >> 538    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:69: error: use of undeclared identifier 'it'
     539        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     540                                                                        ^
  >> 541    /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/Viewer/libViewer/src/DirectoryHa
            ndler.cpp:236:73: error: use of undeclared identifier 'eod'
     542        BOOST_FOREACH (boost::filesystem::path const& p, std::make_pair(it, eod)) {
     543                                                                            ^
     544    fatal error: too many errors emitted, stopping now [-ferror-limit=]
     545    20 errors generated.
  >> 546    make[2]: *** [Viewer/libViewer/src/CMakeFiles/viewer.dir/DirectoryHandler.cpp.o] Error 1
  >> 547    make[1]: *** [Viewer/libViewer/src/CMakeFiles/viewer.dir/all] Error 2
     548    make[1]: *** Waiting for unfinished jobs....
     549    [  4%] Building CXX object ACore/CMakeFiles/core.dir/src/Passwd.cpp.o
     550    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /Users/hei
            nzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/spack/lib/spack/env/clang/clang++ -DCMAKE -DECFLOW_LOGVIEW -DECFLOW_QT5 -DECF_OPENSSL -DQT_NO_KEYWORDS -I/Users/heinzell/work/spa
            ck-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/cereal/include -I/Users/heinzell/work/spack-stack/update
            _spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src -I/Users/heinzell/work/spack-stack/update_spack_stack_from_sp
            ack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7 -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spa
            ck-stack-merge-in-spack-develop/envs/unified-env/install/apple-clang/13.1.6/openssl-3.3.0-56mxtnq/include -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/envs/unifie
            d-env/install/apple-clang/13.1.6/boost-1.85.0-xputgpa/include -DBOOST_NO_CXX98_FUNCTION_BASE -pipe -O3 -DNDEBUG -std=gnu++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX.sdk -mmacosx-version-min=12.0 -fPIC -MD -MT ACore/CMakeFiles/core.dir/src/Passwd.cpp.o -MF CMakeFiles/core.dir/src/Passwd.cpp.o.d -o CMakeFiles/core.dir/src/Passwd.cpp.o -c /Users/heinzell/work/spack-stack/update_spack_sta
            ck_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src/Passwd.cpp
     551    [  4%] Building CXX object ACore/CMakeFiles/core.dir/src/PasswdFile.cpp.o
     552    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /Users/hei
            nzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/spack/lib/spack/env/clang/clang++ -DCMAKE -DECFLOW_LOGVIEW -DECFLOW_QT5 -DECF_OPENSSL -DQT_NO_KEYWORDS -I/Users/heinzell/work/spa
            ck-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/cereal/include -I/Users/heinzell/work/spack-stack/update
            _spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src -I/Users/heinzell/work/spack-stack/update_spack_stack_from_sp
            ack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7 -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spa
            ck-stack-merge-in-spack-develop/envs/unified-env/install/apple-clang/13.1.6/openssl-3.3.0-56mxtnq/include -I/Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/envs/unifie
            d-env/install/apple-clang/13.1.6/boost-1.85.0-xputgpa/include -DBOOST_NO_CXX98_FUNCTION_BASE -pipe -O3 -DNDEBUG -std=gnu++17 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs
            /MacOSX.sdk -mmacosx-version-min=12.0 -fPIC -MD -MT ACore/CMakeFiles/core.dir/src/PasswdFile.cpp.o -MF CMakeFiles/core.dir/src/PasswdFile.cpp.o.d -o CMakeFiles/core.dir/src/PasswdFile.cpp.o -c /Users/heinzell/work/spack-stack/upda
            te_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-src/ACore/src/PasswdFile.cpp
     553    [  4%] Building CXX object ACore/CMakeFiles/core.dir/src/Pid.cpp.o

     ...

     581    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /opt/homeb
            rew/Cellar/cmake/3.27.0/bin/cmake -P CMakeFiles/core.dir/cmake_clean_target.cmake
     582    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /opt/homeb
            rew/Cellar/cmake/3.27.0/bin/cmake -E remove /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twy
            o/spack-build-pmiihk7/ACore/libcore.a
     583    cd /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-pmiihk7/ACore && /opt/homeb
            rew/Cellar/cmake/3.27.0/bin/cmake -E cmake_link_script CMakeFiles/core.dir/link.txt --verbose=1
     584    /usr/bin/ar qc libcore.a CMakeFiles/core.dir/src/AssertTimer.cpp.o CMakeFiles/core.dir/src/Cal.cpp.o CMakeFiles/core.dir/src/Calendar.cpp.o CMakeFiles/core.dir/src/Child.cpp.o CMakeFiles/core.dir/src/CommandLine.cpp.o CMakeFiles/c
            ore.dir/src/DState.cpp.o CMakeFiles/core.dir/src/DurationTimer.cpp.o CMakeFiles/core.dir/src/Ecf.cpp.o CMakeFiles/core.dir/src/Extract.cpp.o CMakeFiles/core.dir/src/File.cpp.o CMakeFiles/core.dir/src/File_r.cpp.o CMakeFiles/core.d
            ir/src/Host.cpp.o CMakeFiles/core.dir/src/Indentor.cpp.o CMakeFiles/core.dir/src/Log.cpp.o CMakeFiles/core.dir/src/LogVerification.cpp.o CMakeFiles/core.dir/src/NOrder.cpp.o CMakeFiles/core.dir/src/NState.cpp.o CMakeFiles/core.dir
            /src/NodePath.cpp.o CMakeFiles/core.dir/src/Passwd.cpp.o CMakeFiles/core.dir/src/PasswdFile.cpp.o CMakeFiles/core.dir/src/Pid.cpp.o CMakeFiles/core.dir/src/PrintStyle.cpp.o CMakeFiles/core.dir/src/SState.cpp.o CMakeFiles/core.dir/
            src/Str.cpp.o CMakeFiles/core.dir/src/StringSplitter.cpp.o CMakeFiles/core.dir/src/TimeSeries.cpp.o CMakeFiles/core.dir/src/TimeSlot.cpp.o CMakeFiles/core.dir/src/TimeStamp.cpp.o CMakeFiles/core.dir/src/User.cpp.o CMakeFiles/core.
            dir/src/Version.cpp.o CMakeFiles/core.dir/src/WhiteListFile.cpp.o
     585    /usr/bin/ranlib libcore.a
     586    [  5%] Built target core
  >> 587    make: *** [all] Error 2

See build log for details:
  /Users/heinzell/work/spack-stack/update_spack_stack_from_spack_dev_20240509/spack-stack-merge-in-spack-develop/cache/build_stage/spack-stage-ecflow-5.11.4-pmiihk7b5bd7f6vnj6twlyddxlh3twyo/spack-build-out.txt

```

### Information on your system

```
* **Spack:** 0.22.0.dev0 (86d6f90a1d79050c65de98eb72b1e239849b24da)
* **Python:** 3.8.9
* **Platform:** darwin-monterey-m1
* **Concretizer:** clingo
```

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/15274473/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/15274474/spack-build-out.txt)

@AlexanderRichert-NOAA and myself

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/ecflow/package.py
+++ b/var/spack/repos/builtin/packages/ecflow/package.py
@@ -52,7 +52,8 @@ class Ecflow(CMakePackage):
     # See https://github.com/spack/spack/pull/22303 for reference
     depends_on(Boost.with_default_variants, when="@:4")
 
-    # Use newer boost with v5
+    # Use newer boost with v5 up to 1.84.0 - https://github.com/spack/spack/issues/44116
+    #conflicts("boost@1.85:", when="@:5.11.4")
     depends_on(
         "boost@1.72:+chrono+date_time+exception+filesystem+program_options+python+regex+serialization+system+test+thread+timer",  # noqa
         when="@5:",

--- a/var/spack/repos/builtin/packages/ecflow/package.py
+++ b/var/spack/repos/builtin/packages/ecflow/package.py
@@ -53,7 +53,7 @@ class Ecflow(CMakePackage):
     depends_on(Boost.with_default_variants, when="@:4")
 
     # Use newer boost with v5 up to 1.84.0 - https://github.com/spack/spack/issues/44116
-    #conflicts("boost@1.85:", when="@:5.11.4")
+    conflicts("boost@1.85:", when="@:5.11.4")
     depends_on(
         "boost@1.72:+chrono+date_time+exception+filesystem+program_options+python+regex+serialization+system+test+thread+timer",  # noqa
         when="@5:",

